clients/src/main/java/org/apache/kafka/common/errors/RecordDeserializationException.java

```diff
@@ -16,29 +16,38 @@
  */
 package org.apache.kafka.common.errors;
 
+import org.apache.kafka.clients.consumer.ConsumerRecord;
 import org.apache.kafka.common.TopicPartition;
 
+
 /**
  *  This exception is raised for any error that occurs while deserializing records received by the consumer using 
  *  the configured {@link org.apache.kafka.common.serialization.Deserializer}.
  */
 public class RecordDeserializationException extends SerializationException {
 
-    private static final long serialVersionUID = 1L;
+    private static final long serialVersionUID = 2L;
     private final TopicPartition partition;
-    private final long offset;
+    private final ConsumerRecord<byte[], byte[]> consumerRecord;
 
-    public RecordDeserializationException(TopicPartition partition, long offset, String message, Throwable cause) {
+    public RecordDeserializationException(TopicPartition partition,
+                                          ConsumerRecord<byte[], byte[]> record,
+                                          String message,
+                                          Throwable cause) {
```

Comment:
IIUC, @mjsax mentioned on the mailing list that we need to keep the existing constructor signature as is and add an overloaded version that accepts the `ConsumerRecord`. Although I'm not sure why... 🤔

Reply:
I have added the existing constructor. I am not convinced by the utility as users are not supposed to instantiate such exceptions.

Reply:
> Although I'm not sure why... 🤔

Because of backward compatibility. `RecordDeserializationException` is a public class, and thus, users could create objects (even if this class is not really intended for user to create objects, but usually only catch this exception -- users do all kind of things -- could also be some test code they write...), and thus we could break there code if we don't keep the old constructor.